libparsec/crates/client/src/workspace/store/mod.rs

```diff
@@ -562,4 +573,65 @@ impl WorkspaceStore {
             .await
             .map_err(UpdateRealmCheckpointError::Internal)
     }
+
+    pub async fn apply_prevent_sync_pattern(&self) -> Result<(), ApplyPreventSyncPatternError> {
+        let mut storage_guard = self.storage.lock().await;
+        let storage = storage_guard
+            .as_mut()
+            .ok_or(ApplyPreventSyncPatternError::Stopped)?;
+
+        let (pattern, applied) = storage
+            .get_prevent_sync_pattern()
+            .await
+            .map_err(ApplyPreventSyncPatternError::Internal)?;
+        if applied {
+            return Ok(()); // Already applied
+        }
+
+        recursive_apply_prevent_sync_pattern(storage, self.realm_id, &pattern, &self.device)
+            .await?;
+        storage
+            .mark_prevent_sync_pattern_fully_applied(&pattern)
+            .map_err(ApplyPreventSyncPatternError::Internal)
+            .await?;
+        Ok(())
+    }
+}
+
+fn recursive_apply_prevent_sync_pattern(
+    storage: &mut WorkspaceStorage,
+    entry_id: VlobID,
+    pattern: &Regex,
+    device: &LocalDevice,
+) -> Box<dyn Future<Output = Result<(), ApplyPreventSyncPatternError>>> {
+    Box::new(async {
+        let Some(manifest) = storage
+            .get_manifest(entry_id)
+            .map_err(ApplyPreventSyncPatternError::Internal)
+            .await?
+        else {
+            return Ok(());
+        };
+
+        // Try folder first
+        let Ok(children) = LocalFolderManifest::decrypt_and_load(&manifest, &device.local_symkey)
+            .map(|folder| folder.children)
+            .or_else(|_| {
+                // Fallback to workspace.
+                LocalWorkspaceManifest::decrypt_and_load(&manifest, &device.local_symkey)
+                    .map(|workspace| workspace.0.children)
+            })
+        else {
+            // TODO: Should it handle `DataError` in case the decryption failed or something else ?
+            return Ok(()); // Not a folder or workspace
+        };
+
+        // TODO: Apply prevent sync pattern
```

Comment:
I'm currently blocked on that TODO, previously we used a `SyncTransactions.apply_prevent_sync_pattern` to apply the pattern but it don't seems that we use this pattern.

I suppose I should do something like `for_update_sync_local_only`, where I call `manifest.apply_prevent_sync_pattern`:

https://github.com/Scille/parsec-cloud/blob/7bb7e0e1b95ea017b7368cf05af72890748ccbb9/libparsec/crates/client/src/workspace/store/sync_updater.rs#L58-L61